service/controller/app/v1/resource/status/create.go

```diff
@@ -50,11 +50,15 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	if !equals(desiredStatus, key.AppStatus(cr)) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for app %#q", name))
 
-		crCopy := cr.DeepCopy()
-		crCopy.ResourceVersion = chart.GetResourceVersion()
-		crCopy.Status = desiredStatus
+		// Get app CR again to ensure the resource version is correct.
```

Comment:
In line 50, we compare the status of CR with the older instance, and we are fetching the new one after that. Why not moving the fetching CR codes just before line 50?

Reply:
The problem is something else might have updated the CR and incremented the resource version. By getting the latest CR just before the update there is less chance of that happening.

Reply:
@tomahawk28 PTAL